openedx_authz/engine/enforcer.py

```diff
@@ -154,6 +157,46 @@ def configure_enforcer_auto_save_and_load(cls):
 
         cls.configure_enforcer_auto_save(auto_save_policy)
 
+    @classmethod
+    def load_policy_if_needed(cls):
+        """Load policy if the last load timestamp indicates it's needed.
+
+        This method checks if the policy needs to be reloaded comparing
+        the last load timestamp with the last modified timestamp in cache
+        and reloads it if necessary.
+
+        Returns:
+            None
+        """
+        last_modified_timestamp = PolicyCacheControl.get_last_modified_timestamp()
+
+        current_timestamp = time.time()
```

Comment:
Instead of using time.time(), as Dave suggested, we could have sort of a counter to compare between local state (cache - local to the process) vs global state (db - shared by all processes). If the global invalidation counter > local invalidation counter then an invalidation occurred.

Reply:
I think a counter can introduce (probably very rare) race conditions where 2 processes increment to the same value and neither of them gets the other's updates, I'd just do something like a guid for safety.